src/github.com/travis-ci/worker/backend/gce.go

```diff
@@ -644,6 +686,47 @@ func (p *gceProvider) imageForLanguage(language string) (*compute.Image, error)
 	return imagesByName[imageNames[len(imageNames)-1]], nil
 }
 
+func (p *gceProvider) imageForLanguage(language string) (*compute.Image, error) {
+	return p.imageByFilter(fmt.Sprintf(gceImageTravisCIPrefixFilter, language))
+}
+
+func (p *gceProvider) imageSelect(ctx gocontext.Context, startAttributes *StartAttributes) (*compute.Image, error) {
+	p.buildImageSelector()
```

Comment:
Forgot to add: One way that looks like it would resolve a potential race condition is moving this to the "initializer".